src/Symfony/Component/HttpFoundation/HeaderUtils.php

```diff
@@ -257,10 +257,20 @@ private static function groupParts(array $matches, string $separators, bool $fir
         $i = 0;
         $partMatches = [];
         $previousMatchWasSeparator = false;
-        foreach ($matches as $match) {
+        foreach ($matches as $key => $match) {
             if (!$first && $previousMatchWasSeparator && isset($match['separator']) && $match['separator'] === $separator) {
                 $previousMatchWasSeparator = true;
                 $partMatches[$i][] = $match;
+            } elseif (!$first && !isset($match['separator']) && isset($matches[$key + 1], $matches[$key + 1]['separator']) && $matches[$key+1]['separator'] === $separator && !isset($matches[$key+2])) { // ends with a separator, so it must be part of the value
+                $numberOfSeparators = 0;
+                for ($i = $key+1; $i < \count($matches); ++$i) {
```

Comment:
do we need this loop? didn't we rule out with `!isset($matches[$key+2])`?

Reply:
Yes we rule out, but we don't know if one or 2 trailing `=`

I agree, its very opaque and I really don't like the whole code :-(